### ISSUE TITLE
[BUG]: Background process of ultralytics_runner consumes a significant portion of CPU after script execution.

### ISSUE THREAD

**Issue by angelusbr** [serengil/deepface#1394]
### Before You Report a Bug, Please Confirm You Have Done The Following...

- [X] I have updated to the latest version of the packages.
- [x] I have searched for both [existing issues](https://github.com/serengil/deepface/issues) and [closed issues](https://github.com/serengil/deepface/issues?q=is%3Aissue+is%3Aclosed) and found none that matched my issue.

### DeepFace's version

0.0.93

### Python version

3.10.12

### Operating System

Ubuntu 22.04.5 LTS

### Dependencies

absl-py==2.1.0
astunparse==1.6.3
beautifulsoup4==4.12.3
blinker==1.9.0
certifi==2024.8.30
charset-normalizer==3.4.0
click==8.1.7
contourpy==1.3.1
cycler==0.12.1
deepface==0.0.93
filelock==3.16.1
fire==0.7.0
Flask==3.1.0
Flask-Cors==5.0.0
flatbuffers==24.3.25
fonttools==4.55.1
fsspec==2024.10.0
gast==0.6.0
gdown==5.2.0
google-pasta==0.2.0
grpcio==1.68.1
gunicorn==23.0.0
h5py==3.12.1
idna==3.10
itsdangerous==2.2.0
Jinja2==3.1.4
joblib==1.4.2
keras==3.7.0
kiwisolver==1.4.7
libclang==18.1.1
lz4==4.3.3
Markdown==3.7
markdown-it-py==3.0.0
MarkupSafe==3.0.2
matplotlib==3.9.3
mdurl==0.1.2
ml-dtypes==0.4.1
mpmath==1.3.0
mtcnn==1.0.0
namex==0.0.8
networkx==3.4.2
numpy==2.0.2
nvidia-cublas-cu12==12.4.5.8
nvidia-cuda-cupti-cu12==12.4.127
nvidia-cuda-nvrtc-cu12==12.4.127
nvidia-cuda-runtime-cu12==12.4.127
nvidia-cudnn-cu12==9.1.0.70
nvidia-cufft-cu12==11.2.1.3
nvidia-curand-cu12==10.3.5.147
nvidia-cusolver-cu12==11.6.1.9
nvidia-cusparse-cu12==12.3.1.170
nvidia-nccl-cu12==2.21.5
nvidia-nvjitlink-cu12==12.4.127
nvidia-nvtx-cu12==12.4.127
opencv-python==4.10.0.84
opt_einsum==3.4.0
optree==0.13.1
packaging==24.2
pandas==2.2.3
pillow==11.0.0
protobuf==5.29.1
psutil==6.1.0
py-cpuinfo==9.0.0
Pygments==2.18.0
pyparsing==3.2.0
PySocks==1.7.1
python-dateutil==2.9.0.post0
pytz==2024.2
PyYAML==6.0.2
requests==2.32.3
retina-face==0.0.17
rich==13.9.4
scipy==1.14.1
seaborn==0.13.2
six==1.17.0
soupsieve==2.6
sympy==1.13.1
tensorboard==2.18.0
tensorboard-data-server==0.7.2
tensorflow==2.18.0
tensorflow-io-gcs-filesystem==0.37.1
termcolor==2.5.0
tf_keras==2.18.0
torch==2.5.1
torchvision==0.20.1
tqdm==4.67.1
triton==3.1.0
typing_extensions==4.12.2
tzdata==2024.2
ultralytics==8.3.41
ultralytics-thop==2.0.12
urllib3==2.2.3
Werkzeug==3.1.3
wrapt==1.17.0

### Reproducible example

```Python
from deepface import DeepFace

a = DeepFace.extract_faces("img1.jpg", detector_backend = "yolov8")
print(a, "< yolov8")
```

### Expected Result

ultralytics_runner should be closed once the Python script is no longer running.

### What happened instead?

ultralytics_runner continues to run in the background after the script has finished, consuming a significant portion of the CPU.

### Additional Info

![image](https://github.com/user-attachments/assets/8aef9797-5dc2-43ae-876f-282ee9ac6893)

Additionally, it seems this issue occurs after the cached_models[task][model_name] = model() line. I tested it using a breakpoint.

![image](https://github.com/user-attachments/assets/eca38c23-937e-414f-acd0-d5d8e7f466bd)

**Comment by serengil** [comment]
That is not a bug, that is in purpose. 

Model is cached not to spend time to build it in next call.

**Comment by bellick100** [comment]
https://github.com/ultralytics/ultralytics/issues/18027

**Comment by zopieux** [comment]
Ultralytics was attacked (or did it on purpose, waiting for a post mortem there), 8.3.41 contains nefarious code downloading and running a crypto miner hosted as a GitHub blob.

All parties were already notified (GitHub, PyPi, ultralytics). My [security advisory](https://github.com/ultralytics/ultralytics/security/advisories/GHSA-36wh-cc56-8jm9#advisory-comment-115615) is pending public release.

**Comment by angelusbr** [comment]
Exactly. That's why this process continues running after I close VSCode.

Em qui., 5 de dez. de 2024 08:42, Alexandre Macabies <
***@***.***> escreveu:
